### ISSUE TITLE
SPMM 

### ISSUE THREAD

**Issue by guohh516** [google-research/sputnik#14]
When initializing the `sparse_tile_loader`，the threadIdx.x should be threadIdx.x%kBlockWidth. Is what I said correct ?

**Comment by tgale96** [comment]
Hi! We pass `threadIdx.x` directly ([code](https://github.com/google-research/sputnik/blob/master/sputnik/spmm/cuda_spmm.cu.cc#L164)).

**Comment by guohh516** [comment]
But if using `subwarp tiling`，different subwarps should correspond to a new line？

**Comment by tgale96** [comment]
I believe that is handled in the block configuration passed in for kernel launch.

**Comment by guohh516** [comment]
Should I use CudaSpmm directly to use the lib after I make install? [code](https://github.com/google-research/sputnik/blob/bbf5840ba5efccf01f862855c785f71bcc6ff1f0/sputnik/spmm/cuda_spmm.h#L51)

**Comment by guohh516** [comment]
And if I don't use `bias`, what I need to do is just pass `nullptr` to this argument or use the default value, is what i said correct?

**Comment by tgale96** [comment]
Yes, CudaSpmm is the right API. If you don't need a fused bias + relu you can call this API. If you want to fuse the operations we have CudaSpmmBiasRelu.

**Comment by guohh516** [comment]
Emmm   I have a question. This project determines the configuration of spmmconfig based on the size of the input dense matrix, but this introduces runtime on the CPU. The time I spend directly using `cudaspmm` is much longer than cuSPARSE, but if I refactor the project and configure `spmmconfig` myself, the time is shorter than cuSPARSE. But in this case, does it mean that the universality is not good enough, or do I suggest create some instances of `cudaspmmex` specific to `spmmconfig` in the library? Is this feasible?

**Comment by tgale96** [comment]
Interesting! I would think your problem must be quite small for that to be the case? The tuning heuristics in this library are by no means expected to be good across all problems and if you know what config is best for your problem you should pass that explicitly.
